mne/time_frequency/tests/test_csd.py

```diff
@@ -223,17 +224,25 @@ def test_csd_get_data():
 @requires_version('h5io')
 def test_csd_save(tmp_path):
     """Test saving and loading a CrossSpectralDensity."""
-    csd = _make_csd()
-    tempdir = str(tmp_path)
-    fname = op.join(tempdir, 'csd.h5')
+    raw = mne.io.read_raw_fif(raw_fname)
+    events = mne.find_events(raw)
+    picks = mne.pick_types(raw.info, meg='grad')
+    projs = compute_proj_raw(raw, n_grad=1, n_mag=1, n_eeg=1, reject=None)
+    raw.add_proj(projs)
+    epochs = mne.Epochs(raw, events, event_id=1, tmin=-0.2, tmax=1,
+                        picks=picks, baseline=(None, 0),
+                        reject=dict(grad=4000e-13), preload=True)
+    csd = csd_fourier(epochs, fmin=15, fmax=20)
+    fname = op.join(str(tmp_path), 'csd.h5')
     csd.save(fname)
     csd2 = read_csd(fname)
     assert_array_equal(csd._data, csd2._data)
+    assert_array_equal(csd.frequencies, csd2.frequencies)
     assert csd.tmin == csd2.tmin
     assert csd.tmax == csd2.tmax
     assert csd.ch_names == csd2.ch_names
-    assert csd.frequencies == csd2.frequencies
     assert csd._is_sum == csd2._is_sum
+    assert isinstance(csd2.projs[0], Projection)
```

Comment:
how long does this test take to run? I am a bit worried we have to go from raw, epochs to test IO of a container.

Reply:
It does slow down the test, on my macbook air:
- Before 'pytest test_csd.py' took ~5 s.
- After the changes it took ~9 s.

Reply:
You could try adding a parameter to `_make_csd(add_proj=False)` and use `add_proj=True` here. If you need help figuring out how to add a projector let me know

Reply:
(then you don't need to actually use raw/epochs, you can use whatever `make_stc` is doing)

Reply:
This _make_stc? https://github.com/mne-tools/mne-python/blob/main/mne/source_estimate.py#L388-L430.

Adding a add_proj argument to _make_csd make sense to me. I wasn't sure how to create the Projections object - the docs for Projections suggests using a compute_proj_* function.

Reply:
How about I use `mne.proj.read_proj` with `io/tests/data/test-proj.fif`?

Reply:
Just tried `read_proj` with `io/tests/data/test-proj.fif`, works well I think. The test duration is the same before and after the changes (~5 s). Shall I commit the change? Or do you have a better suggestion for generating the Projection?

Reply:
That won't work because the names won't match. This should be close:
```
if add_proj:
    proj_data = dict(col_names=None, row_names=ch_names, data=np.ones((len(ch_names), 1)))
    proj = Projection(proj_data)
else:
    proj = None
... = CrossSpectralDensity(..., proj=proj)
```

Reply:
Alright, thanks. The latest commit should resolve this.